neps/nep-0568.md

```diff
@@ -73,6 +73,38 @@ post-processing, as long as the chain's view reflects a fully resharded state.
 
 ### State Storage - Flat State
 
+Flat State is a collection of key-value pairs stored on disk and each entry
+contains a reference to its ShardId. When splitting a shard, every item inside
+its Flat State must be correctly reassigned to either one of the new children;
+due to technical limitations such operation can't be completed instantaneously.
+
+Flat State main purposes are allowing the creation of State Sync snapshots and
+the construction of Mem Tries. Fortunately, these two operations can be delayed
+until resharding is completed. Note also that with Mem Tries enabled the chain
+can move forward even if the current status of Flat State is not in sync with
+the latest block. 
+
+For the reason stated above, the chosen strategy is to reshard Flat State in a
+long-running background task. The new shards' states must converge with their
+Mem Tries representation in a reasonable amount of time, indicatively by the
+first half of the new epoch.
```

Comment:
Why the first half of the new epoch? It's just for my information, no need to add it here. I don't think it's too important so maybe skip this part (the "indicatively by the first half of the new epoch").

Reply:
Removed!

It was badly worded; the flat state must converge with memtrie, at some point in time.

The time consideration was to have flat storage ready soon enough so that state parts for the new shards can be distributed before the next epoch.